src/libraries/System.Private.CoreLib/src/System/IO/FileInfo.cs

```diff
@@ -200,5 +200,12 @@ public FileInfo Replace(string destinationFileName, string? destinationBackupFil
 
         [SupportedOSPlatform("windows")]
         public void Encrypt() => File.Encrypt(FullPath);
+
+        private StreamWriter CreateStreamWriter(bool append)
+        {
+            StreamWriter streamWriter = new StreamWriter(NormalizedPath);
```

Comment:
we need to pass `append` argument to the `StreamWriter` ctor:

```suggestion
            StreamWriter streamWriter = new StreamWriter(NormalizedPath, append);
```